src/sage/graphs/graph_decompositions/slice_decomposition.pyx

```diff
@@ -0,0 +1,970 @@
+        """
+        l = self.xslice_len[idx]
+        S = [ [self.sigma[idx]] ]
+        j = idx + 1
+        while j < idx + l:
+            lj = self.xslice_len[j]
+            S.append(list(self.sigma[j:j+lj]))
+            j += lj
+        assert j == idx + l, "slice decomposition is ill-formed"
+        return S
+
+    def _xslice_lex_label(self, idx):
+        r"""
+        This method is for internal use only
+        """
+        return list(self.lex_label[idx])
+
+    def _xslice_active_edges(self, idx):
+        r"""
+        This method is for internal use only
+        """
+        l = self.xslice_len[idx]
+        llv_prefix = len(self.lex_label[idx])
+        A = []
+        j = idx + 1
+        while j < idx + l:
+            lj = self.xslice_len[j]
+            llj = self.lex_label[j]
+            for u in self.sigma[j:j+lj]:
+                for w in llj[llv_prefix:]:
+                    A.append((w, u))
+            j += lj
+        assert j == idx + l, "slice decomposition is ill-formed"
+        return A
+
+    def underlying_graph(self):
+        r"""
+        Return the underlying graph corresponding to the slice decomposition.
+
+        If `G` was the graph given as parameter to compute the slice
+        decomposition, the underlying graph corresponds to ``G.to_simple()``,
+        *i.e.*, it is the input graph without loops and multiple edges.
+
+        EXAMPLES:
+
+        ::
+
+            sage: G = Graph('L~mpn~Nrv{^o~_').relabel('abcdefguvwxyz',inplace=False)
+            sage: SD = G.slice_decomposition(initial_vertex='x')
+            sage: SD.underlying_graph() == G
+            True
+
+        The graph can have loops or multiple edges but they are ignored::
+
+            sage: G = graphs.CubeConnectedCycle(2) # multiple edges
+            sage: SD = G.slice_decomposition()
+            sage: SD.underlying_graph() == G.to_simple(immutable=True)
+            True
+
+            sage: G = graphs.CubeConnectedCycle(1) # loops
+            sage: SD = G.slice_decomposition()
+            sage: SD.underlying_graph() == G.to_simple(immutable=True)
+            True
+
+        TESTS:
+
+            sage: for _ in range(5):
+            ....:   G = graphs.RandomGNP(15, 0.3)
+            ....:   SD = G.slice_decomposition()
+            ....:   SD.underlying_graph() == G
+            True
+            True
+            True
+            True
+            True
+        """
+        if not hasattr(self, '_underlying_graph'):
+            vertices = self.sigma
+            edges = [(u, v) for i, v in enumerate(self.sigma)
+                            for u in self.lex_label[i]]
+            data = [vertices, edges]
+            Gclass = self._graph_class
+            self._underlying_graph = Gclass(data, format='vertices_and_edges',
+                                            immutable=True)
+        return self._underlying_graph
+
+    def _repr_(self):
+        def inner_repr(idx):
```

Comment:
add 1 line description + doctest.

Reply:
done in 1fbb00ebdb1de31ab09729fe20f6e949acefd770